aggregator/operator_registrations_inmemory.go

```diff
@@ -0,0 +1,190 @@
+package aggregator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/avsregistry"
+	"github.com/Layr-Labs/eigensdk-go/crypto/bls"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/services/operatorpubkeys"
+	"github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type OperatorRegistrationsService interface {
+	operatorpubkeys.OperatorPubkeysService
+
+	GetOperatorPubkeysById(ctx context.Context, operatorId types.OperatorId) (operatorPubkeys types.OperatorPubkeys, operatorFound bool)
+}
+
+type OperatorRegistrationsServiceInMemory struct {
+	avsRegistrySubscriber avsregistry.AvsRegistrySubscriber
+	avsRegistryReader     avsregistry.AvsRegistryReader
+	logger                logging.Logger
+	queryByAddrC          chan<- queryByAddr
+	queryByIdC            chan<- queryById
+}
+
+type queryByAddr struct {
+	operatorAddr common.Address
+	// channel through which to receive the response (operator pubkeys)
+	respC chan<- resp
+}
+type queryById struct {
+	operatorId types.OperatorId
+	// channel through which to receive the response (operator pubkeys)
+	respC chan<- resp
+}
+
+type resp struct {
+	operatorPubkeys types.OperatorPubkeys
+	// false if operators were not present in the pubkey dict
+	operatorExists bool
+}
+
+var _ operatorpubkeys.OperatorPubkeysService = (*OperatorRegistrationsServiceInMemory)(nil)
+
+// NewOperatorRegistrationsServiceInMemory constructs a OperatorRegistrationsServiceInMemory and starts it in a goroutine.
+// It takes a context as argument because the "backfilling" of the database is done inside this constructor,
+// so we wait for all past NewPubkeyRegistration events to be queried and the db to be filled before returning the service.
+// The constructor is thus following a RAII-like pattern, of initializing the serving during construction.
+// Using a separate initialize() function might lead to some users forgetting to call it and the service not behaving properly.
+func NewOperatorRegistrationsServiceInMemory(
+	ctx context.Context,
+	avsRegistrySubscriber avsregistry.AvsRegistrySubscriber,
```

Comment:
IMO we should send a SafeClient and the BLSRegistry address here, create the bindings in our own and manage them here. Reason for that is it's just simpler, we don't seem to have a good reason to use a separate module. Or we should at least make sure we are using a SafeClient when instantiating this so we can rely on it for resubs and such.

Reply:
Done in a slightly different way:
Using our AvsSubscriber instead which uses SafeClient
AvsSubscriber now implements AvsRegistrySubscriber interface to be able to work with registry

Reply:
I see that we're using the eigensdk `AvsRegistrySubscriber` rather than our own `AvsSubscriber`. Is the comment outdated?

Reply:
Not really. AvsSubscriber implements methods of AvsRegistrySubscriber. 

From`aggregator` we pass AvsSubscriber into NewOperatorRegistrationsServiceInMemory. Since `interface AvsSubscriber: AvsRegistrySubscriber` we can and use `AvsRegistrySubscriber` here directly since we need only its methods.

Reply:
This enables use of this struct with both our AvsSubscriber and el AvsRegistrySubscriber, in case one wants to